doubles/allowance.py

```diff
@@ -119,6 +120,17 @@ def with_args(self, *args, **kwargs):
         self.verify_arguments()
         return self
 
+    def with_args_that_match(self, matching_function):
```

Comment:
`with_args_callback`? Not sure if that's better, but match seems to imply something like a regex. I'll try to think about it.